Broke on LibreSSL 2.1.4

```
tlsc.c:158:3: warning: implicit declaration of function 'tls_config_insecure_noverifyhost' is invalid in C99
      [-Wimplicit-function-declaration]
                tls_config_insecure_noverifyhost(tls_config);
                ^
cc -o socks socks.o
1 warning generated.
cc -o tlsc tlsc.o -lcrypto `pkg-config --libs libtls libssl`
Undefined symbols for architecture x86_64:
  "_tls_config_insecure_noverifyhost", referenced from:
      _main in tlsc.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make: *** [tlsc] Error 1
```

The problem is that in the LibreSSL 2.1.4 release they renamed the:

```
void tls_config_insecure_noverifyhost(struct tls_config *config);
void tls_config_insecure_noverifycert(struct tls_config *config);
```

Block your code is looking for in the ` tls.h ` header. It [seems](http://www.freshbsd.org/commit/openbsd/2b22762d1139c74c743195f46b41fea0b9459ecd) to have [become](https://github.com/libressl-portable/portable/commit/9da890ffb97c1ae9699e74666fb7186308435973):

` tls_config_insecure_noverifyname `

--- a/tlsc.c
+++ b/tlsc.c
@@ -65,7 +65,7 @@ main(int argc, char *argv[], char *envp[])
 	char *ca_path = NULL;
 	char *cert_file = NULL;
 	bool no_verification = false;
-	bool no_host_verification = false;
+	bool no_name_verification = false;
 	bool no_cert_verification = false;
 	char *host = getenv("TCPREMOTEHOST");
 	struct tls_config *tls_config;
@@ -85,7 +85,7 @@ main(int argc, char *argv[], char *envp[])
 			if ((host = strdup(optarg)) == NULL) goto err;
 			break;
 		case 'H':
-			no_host_verification = true;
+			no_name_verification = true;
 			break;
 		case 'C':
 			no_cert_verification = true;
@@ -154,8 +154,8 @@ main(int argc, char *argv[], char *envp[])
 	if (no_cert_verification)
 		tls_config_insecure_noverifycert(tls_config);
 
-	if (no_host_verification)
-		tls_config_insecure_noverifyhost(tls_config);
+	if (no_name_verification)
+		tls_config_insecure_noverifyname(tls_config);
 
 	if (no_verification)
 		tls_config_insecure_noverifycert(tls_config);

